dallinger/experiment_server/sockets.py

```diff
@@ -114,14 +147,23 @@ def unsubscribe(self, client):
 class Client(object):
     """Represents a single websocket client."""
 
-    def __init__(self, ws, lag_tolerance_secs=0.1):
+    def __init__(self, ws, lag_tolerance_secs=0.1, worker_id=None, participant_id=None):
         self.ws = ws
         self.lag_tolerance_secs = lag_tolerance_secs
+        self.worker_id = worker_id
+        self.participant_id = participant_id
 
         # This lock is used to make sure that multiple greenlets
         # cannot send to the same socket concurrently.
         self.send_lock = Semaphore()
 
+    def client_info(self):
+        return {
+            "class": self.__class__.__module__ + "." + self.__class__.__module__,
+            "worker_id": self.worker_id,
+            "participant_id": self.participant_id,
+        }
+
```

Comment:
Do you think we should check that the `worker_id` and `participant_id` are a valid combination (by looking at the database)? Might be good for security?

Reply:
We don't do that sort of thing in other parts of Dallinger, so I'm not sure we need to do so here here. My thought was that this was mostly and either/or situation where `participant_id` would generally be preferred, but if for some reason the connection was established before a `participant_id` was assigned then the experiment might be able to make use of the `worker_id` instead. This information isn't used in Dallinger by default, it would be up to the experiment developer to decide how they might want to use it.